docs/block-diagram.md

```diff
@@ -0,0 +1,40 @@
+# Block Diagram
+
+## Introduction
+
+This document provides a concise block diagram illustrating the architecture of our software application, Osmo-Notify. The diagram outlines the key components and their interactions to help you better understand how the application functions.
+
+## Block Diagram Overview
+
+The block diagram represents the software application's architecture in a visual format, making it easier to grasp its structure and functionality.
+
+## Block Diagram
+
+![Block Diagram](./assets/osmo-notify-block-diagram.png)
+
+## Components and Functionalities
+
+### 1. User/Application
+
+This component is responsible for making the request to the Osmo-Notify API with required details about the notification.
+
+### 2. API
```

Comment:
Diagram is a bit incorrect.

APIs Job is just one part. It takes the requests from users and adds details to DBs and gives confirmation in response. Its disconnected from other parts like CRON, Queues. It doesn't communicate with CRON or Queues.

CRON fetches from DB and Adds pending notifications to Queues (redis Queue via Bull MQ) which again communicate to respective services(consumers), and updates the DB again with response from the provider service.

I would say make a block inside osmo-notify again so that it shows sepration of these two process.
1. API process.
2. CRON Process.

DB can be common anyway.

Reply:
This has been resolved in the commit https://github.com/OsmosysSoftware/osmo-notify/pull/63/commits/cfbb593260b078a9ab1fb4c68223598a576f424a